go/host/rpc/clientapi/client_api_eth.go

```diff
@@ -48,21 +48,21 @@ func (api *EthereumAPI) BlockNumber() hexutil.Uint64 {
 }
 
 // GetBlockByNumber returns the header of the batch with the given height.
-func (api *EthereumAPI) GetBlockByNumber(ctx context.Context, number rpc.BlockNumber, _ bool) (common.BatchHeader, error) {
+func (api *EthereumAPI) GetBlockByNumber(ctx context.Context, number rpc.BlockNumber, _ bool) (*common.BatchHeader, error) {
```

Comment:
it was needed to change the return to a pointer so that it uses the custom json unmarshaller defined on (*BatchHeader)